packages/plugin-react/src/index.ts

```diff
@@ -174,6 +185,11 @@ export default function viteReact(opts: Options = {}): PluginOption[] {
     config: () => ({
       resolve: {
         dedupe: ['react', 'react-dom']
+      },
+      // react/jsx-runtime is in CJS format
+      // we want to explicitly cast it to ESM
+      optimizeDeps: {
+        include: ['react/jsx-runtime']
```

Comment:
You sure this is necessary? My project was working fine without it. 🤔

Reply:
I have tested with `packages/playground/react` which now enables `automic` runtime. Without explicitly optimize `react/jsx-runtime`, during runtime the browser tries to load `react/jsx-runtime` which is plain CJS with the following content:
```js
'use strict';

if (process.env.NODE_ENV === 'production') {
  module.exports = require('./cjs/react-jsx-runtime.production.min.js');
} else {
  module.exports = require('./cjs/react-jsx-runtime.development.js');
}
```

I think the issue is that during vite's import scanning phase this is not being seen. I am not sure if there is another way to include it into optimization instead of put it here?

Reply:
Oh it happens in development. That makes sense, as `vite-react-jsx` doesn't apply the transform in development, but this plugin does.